src/librustdoc/html/format.rs

```diff
@@ -460,7 +460,8 @@ impl fmt::Display for clean::Type {
                 f.write_str(name)
             }
             clean::ResolvedPath{ did, ref typarams, ref path } => {
-                try!(resolved_path(f, did, path, false));
+                // Paths like Self::Output should be rendered with all segments
```

Comment:
(In fact, I believe a hybrid path like `std::slice::Iter<T>::Item` is possible too.)